module/VuFind/src/VuFind/Db/Service/LoginTokenService.php

```diff
@@ -173,6 +233,18 @@ public function getBySeries(string $series): array
      */
     public function deleteExpired(DateTime $dateLimit, ?int $limit = null): int
     {
-        return $this->getDbTable('LoginToken')->deleteExpired($dateLimit->format('Y-m-d H:i:s'), $limit);
+        $subQueryBuilder = $this->entityManager->createQueryBuilder();
+        $subQueryBuilder->select('lt.id')
+            ->from($this->getEntityClass(LoginTokenEntityInterface::class), 'lt')
+            ->where('lt.lastLogin < :dateLimit')
```

Comment:
I think expiration needs to be based on the `expires` field, not `lastLogin`. Since that's an integer timestamp, you'll need to compare it against `$dateLimit->getTimestamp()` if I'm remembering correctly.